mantidproject/mantid
# Add d16 loader

**Description of work.**
WIP for #27615 

This PR updates LoadILLSANS so it can load and display data from D16 when it is used in SANS mode, and adds the corresponding IDF and loader tests.
IDF generator can be found in https://github.com/mantidproject/mantidgeometry/pull/162 .

**To test:**
- Run cmake so that the new test data are uploaded
- Build datahandlingtest
- Run `ctest -R LoadILLSANS` to check the unitTest.
 or 
- In Mantid, open `Testing/Data/ILL/D16/218356.nxs` and check everything works fine.
This file is not from the up-to-date version of the nexus files for D16, and as such lacks a field indicating lambda resolution and so raises a warning when loaded. This should be corrected in newer nexus files. 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
